// @nx/storybook/src/executors/storybook/storybook.impl.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = storybookExecutor;
const utilities_1 = require("../../utils/utilities");
const semver_1 = require("semver");
async function* storybookExecutor(options, context) {
    const storybookMajor = (0, utilities_1.storybookMajorVersion)();
    if (storybookMajor > 0 && storybookMajor <= 7) {
        throw (0, utilities_1.pleaseUpgrade)();
    }
    (0, utilities_1.storybookConfigExistsCheck)(options.configDir, context.projectName);
    const buildOptions = options;
    const result = await runInstance(buildOptions);
    yield {
        success: true,
        info: {
            port: result?.['port'],
            baseUrl: `${options.https ? 'https' : 'http'}://${options.host ?? 'localhost'}:${result?.['port']}`,
        },
    };
    await new Promise(() => { });
}
async function runInstance(options) {
    const installedStorybookVersion = (0, utilities_1.getInstalledStorybookVersion)();
    const hasCoreServerInStorybookPackage = (0, semver_1.gte)(installedStorybookVersion, '8.2.0');
    const storybookCore = await (hasCoreServerInStorybookPackage
        ? Promise.resolve().then(() => require('storybook/internal/core-server')) : // This is needed for backwards compatibility - but we do not have the package installed in the nx repo
     Promise.resolve().then(() => require('@storybook/core-server')));
    const env = process.env.NODE_ENV ?? 'development';
    process.env.NODE_ENV = env;
    return storybookCore.build({
        ...options,
        mode: 'dev',
    });
}
