// @nx/vue/src/utils/add-linting.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.addLinting = addLinting;
const devkit_1 = require("@nx/devkit");
const eslint_1 = require("@nx/eslint");
const eslint_file_1 = require("@nx/eslint/src/generators/utils/eslint-file");
const flat_config_1 = require("@nx/eslint/src/utils/flat-config");
const version_utils_1 = require("@nx/eslint/src/utils/version-utils");
const path_1 = require("nx/src/utils/path");
const versions_1 = require("./versions");
const semver_1 = require("semver");
async function addLinting(host, options, projectType) {
    if (options.linter === 'eslint') {
        const tasks = [];
        const lintTask = await (0, eslint_1.lintProjectGenerator)(host, {
            linter: options.linter,
            project: options.projectName,
            tsConfigPaths: [
                (0, path_1.joinPathFragments)(options.projectRoot, `tsconfig.${projectType}.json`),
            ],
            unitTestRunner: options.unitTestRunner,
            skipFormat: true,
            setParserOptionsProject: options.setParserOptionsProject,
            rootProject: options.rootProject,
            addPlugin: options.addPlugin,
        });
        tasks.push(lintTask);
        if ((0, flat_config_1.useFlatConfig)(host)) {
        }
        else {
            const addExtendsTask = (0, eslint_file_1.addExtendsToLintConfig)(host, options.projectRoot, [
                'plugin:vue/vue3-essential',
                'eslint:recommended',
                '@vue/eslint-config-typescript',
                '@vue/eslint-config-prettier/skip-formatting',
            ].filter(Boolean));
            tasks.push(addExtendsTask);
        }
        editEslintConfigFiles(host, options.projectRoot);
        const eslintVersion = (0, version_utils_1.getInstalledEslintVersion)(host);
        const devDependencies = {
            '@vue/eslint-config-prettier': versions_1.vueEslintConfigPrettierVersion,
            '@vue/eslint-config-typescript': eslintVersion && (0, semver_1.lt)(eslintVersion, '9.0.0')
                ? versions_1.vueEslintConfigTypescriptVersion
                : versions_1.eslint9__VueEslintConfigTypescriptVersion,
            'eslint-plugin-vue': versions_1.eslintPluginVueVersion,
        };
        if ((0, eslint_file_1.isEslintConfigSupported)(host, options.projectRoot) &&
            (0, flat_config_1.useFlatConfig)(host)) {
            devDependencies['@typescript-eslint/parser'] =
                (0, version_utils_1.getTypeScriptEslintVersionToInstall)(host);
        }
        if (!options.skipPackageJson) {
            const installTask = (0, devkit_1.addDependenciesToPackageJson)(host, {}, devDependencies);
            tasks.push(installTask);
        }
        return (0, devkit_1.runTasksInSerial)(...tasks);
    }
    else {
        return () => { };
    }
}
function editEslintConfigFiles(tree, projectRoot) {
    const hasVueFiles = (o) => o.files &&
        (Array.isArray(o.files)
            ? o.files.some((f) => f.endsWith('*.vue'))
            : o.files.endsWith('*.vue'));
    const addVueFiles = (o) => {
        if (!o.files) {
            o.files = ['*.vue'];
        }
        else if (Array.isArray(o.files)) {
            o.files.push('*.vue');
        }
        else {
            o.files = [o.files, '*.vue'];
        }
    };
    if ((0, eslint_file_1.isEslintConfigSupported)(tree, projectRoot)) {
        if ((0, flat_config_1.useFlatConfig)(tree)) {
            (0, eslint_file_1.addPredefinedConfigToFlatLintConfig)(tree, projectRoot, 'flat/recommended', { moduleName: 'vue', moduleImportPath: 'eslint-plugin-vue' });
            // This allows .vue files to be parsed
            (0, eslint_file_1.addOverrideToLintConfig)(tree, projectRoot, {
                files: ['**/*.vue'],
                languageOptions: {
                    parserOptions: {
                        parser: '@typescript-eslint/parser',
                    },
                },
            } // languageOptions is not present on eslintrc override, but it is for flat config
            );
            // Add an empty rules object to users know how to add/override rules
            (0, eslint_file_1.addOverrideToLintConfig)(tree, projectRoot, {
                files: ['*.ts', '*.tsx', '*.js', '*.jsx', '*.vue'],
                rules: { 'vue/multi-word-component-names': 'off' },
            });
        }
        else {
            if ((0, eslint_file_1.lintConfigHasOverride)(tree, projectRoot, (o) => o.parserOptions && !hasVueFiles(o), true)) {
                (0, eslint_file_1.updateOverrideInLintConfig)(tree, projectRoot, (o) => !!o.parserOptions, (o) => {
                    addVueFiles(o);
                    return o;
                });
            }
            else {
                (0, eslint_file_1.replaceOverridesInLintConfig)(tree, projectRoot, [
                    {
                        files: ['*.ts', '*.tsx', '*.js', '*.jsx', '*.vue'],
                        rules: { 'vue/multi-word-component-names': 'off' },
                    },
                ]);
            }
        }
    }
    // Edit root config too
    if (!(0, eslint_file_1.isEslintConfigSupported)(tree)) {
        return;
    }
    if ((0, eslint_file_1.lintConfigHasOverride)(tree, '', (o) => o.rules?.['@nx/enforce-module-boundaries'] && !hasVueFiles(o), true)) {
        (0, eslint_file_1.updateOverrideInLintConfig)(tree, '', (o) => !!o.rules?.['@nx/enforce-module-boundaries'], (o) => {
            addVueFiles(o);
            return o;
        });
    }
}
