// @nx/vue/src/utils/versions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.lessVersion = exports.sassVersion = exports.autoprefixerVersion = exports.tailwindcssVersion = exports.postcssVersion = exports.eslintPluginVueVersion = exports.eslint9__VueEslintConfigTypescriptVersion = exports.vueEslintConfigTypescriptVersion = exports.vueEslintConfigPrettierVersion = exports.vitePluginVueVersion = exports.vueTestUtilsVersion = exports.vueRouterVersion = exports.vueTscVersion = exports.vueVersion = exports.nxVersion = void 0;
exports.nxVersion = require('../../package.json').version;
// vue core
exports.vueVersion = '^3.5.13';
exports.vueTscVersion = '^2.2.8';
exports.vueRouterVersion = '^4.5.0';
// test deps
exports.vueTestUtilsVersion = '^2.4.6';
exports.vitePluginVueVersion = '^6.0.1';
// linting deps
exports.vueEslintConfigPrettierVersion = '^10.2.0';
exports.vueEslintConfigTypescriptVersion = '^11.0.3';
exports.eslint9__VueEslintConfigTypescriptVersion = '^14.6.0';
exports.eslintPluginVueVersion = '^9.16.1';
// tailwindcss
exports.postcssVersion = '8.4.21';
exports.tailwindcssVersion = '3.2.7';
exports.autoprefixerVersion = '10.4.13';
// other deps
exports.sassVersion = '^1.70.0';
exports.lessVersion = '3.12.2';
